pyo3-build-config/src/impl_.rs

```diff
@@ -2733,6 +2749,44 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_build_script_outputs_gil_disabled() {
+        let mut build_flags = BuildFlags::default();
+        build_flags.0.insert(BuildFlag::Py_GIL_DISABLED);
+        let interpreter_config = InterpreterConfig {
+            implementation: PythonImplementation::CPython,
+            version: PythonVersion {
+                major: 3,
+                minor: 13,
+            },
+            shared: true,
+            abi3: false,
+            lib_name: Some("python3".into()),
+            lib_dir: None,
+            executable: None,
+            pointer_width: None,
+            build_flags,
+            suppress_build_script_link_lines: false,
+            extra_build_script_lines: vec![],
+        };
+
+        assert_eq!(
+            interpreter_config.build_script_outputs(),
+            [
+                "cargo:rustc-cfg=Py_3_6".to_owned(),
+                "cargo:rustc-cfg=Py_3_7".to_owned(),
+                "cargo:rustc-cfg=Py_3_8".to_owned(),
+                "cargo:rustc-cfg=Py_3_9".to_owned(),
+                "cargo:rustc-cfg=Py_3_10".to_owned(),
+                "cargo:rustc-cfg=Py_3_11".to_owned(),
+                "cargo:rustc-cfg=Py_3_12".to_owned(),
+                "cargo:rustc-cfg=Py_3_13".to_owned(),
+                "cargo:rustc-cfg=Py_GIL_DISABLED".to_owned(),
+                "cargo:rustc-cfg=py_sys_config=\"Py_GIL_DISABLED\"".to_owned(),
```

Comment:
should we define both configs in this case? Or only the top-level `Py_GIL_DISABLED` one?

Reply:
I think let's go for just the top-level `Py_GIL_DISABLED` 👍. (You'll need to special-case this when printing the `BuildFlags`. as cfgs, I guess)

Please document in `pyo3-build-config/lib.rs` (in the docstring for `use_pyo3_cfgs`).